e2e/exchanges_test.go

```diff
@@ -63,67 +63,40 @@ func (s *ExchangesE2ESuite) SetupTest() {
-	err := infestor.NewMocksBuilder().Reset().Add(ex).Deploy(s.api)
-	s.Require().Error(err)
-
-	// With set rate and price
-	price := "0x0000000000000000000000000000000000000000000000000dd22d6848e229b8"
-	ex = ex.WithCustom("rate", price).WithCustom("price", price)
-	err = infestor.NewMocksBuilder().Debug().Reset().Add(ex).Deploy(s.api)
+func (s *ExchangesE2ESuite) TestEthRPC() {
+	const blockNumber int = 100
+	ex := origin.NewExchange("ethrpc").
+		WithCustom("blockNumber", blockNumber).
+		WithCustom("tokens",
+			[]types.Address{
+				types.MustAddressFromHex("0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84"),
+				types.MustAddressFromHex("0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"),
+			}).
+		WithFunctionData("symbols", []origin.FunctionData{
+			{
+				Args:   []any{},
+				Return: []any{"stETH"},
+			},
+			{
+				Args:   []any{},
+				Return: []any{"WETH"},
+			},
+		}).
+		WithFunctionData("decimals", []origin.FunctionData{
+			{
+				Args:   []any{},
+				Return: []any{big.NewInt(18)},
+			},
+			{
+				Args:   []any{},
+				Return: []any{big.NewInt(18)},
+			},
+		})
+	err := infestor.NewMocksBuilder().Reset().Debug().Add(ex).Deploy(s.api)
 	s.Require().NoError(err)
 
 	url := fmt.Sprintf("%s/", s.url)
-	reqJSON := `{"method":"eth_call","params":[{"from":null,"to":"0x0000000000000000000000000000000000000000","data":"0xb10be7390000000000000000000000000000000000000000000000000000000000000000"}, "latest"],"id":1,"jsonrpc":"2.0"}`
+	reqJSON := fmt.Sprintf(origin.RPCCallRequestJSON, "null", "0xeefba1e63905ef1d7acba5a8513c70307c1ce441", "0x252dba42000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001800000000000000000000000000000000000000000000000000000000000000200000000000000000000000000ae7ab96520de3a18e5e111b5eaab095312d7fe840000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000495d89b4100000000000000000000000000000000000000000000000000000000000000000000000000000000ae7ab96520de3a18e5e111b5eaab095312d7fe8400000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000004313ce56700000000000000000000000000000000000000000000000000000000000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc20000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000495d89b4100000000000000000000000000000000000000000000000000000000000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc200000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000004313ce56700000000000000000000000000000000000000000000000000000000", "latest")
```

Comment:
The bytes array being sent to Mock RPC url should be the calldata encoded by multicall structure.